cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/LanguageTraits.kt

```diff
@@ -196,11 +196,22 @@ interface HasGlobalVariables : LanguageTrait
 /**
  * A language trait, that specifies that the language has so-called functional style casts, meaning
  * that they look like regular call expressions. Since we can therefore not distinguish between a
- * [CallExpression] and a [CastExpression], we need to employ an additional pass
- * ([ReplaceCallCastPass]) after the initial language frontends are done.
+ * [CallExpression] and a [CastExpression], we need to employ an additional pass ([ReplaceCallPass])
+ * after the initial language frontends are done.
  */
 interface HasFunctionalCasts : LanguageTrait
 
+/**
+ * A language trait, that specifies that the language has functional style constructs, meaning that
+ * constructor calls look like regular call expressions (usually meaning that the language has no
+ * dedicated `new` keyword).
+ *
+ * Since we can therefore not distinguish between a [CallExpression] and a [ConstructExpression] in
+ * the frontend, we need to employ an additional pass ([ReplaceCallPass]) after the initial language
+ * frontends are done.
+ */
+interface HasFunctionalConstructs : LanguageTrait
```

Comment:
Can we rename this to HasFunctionalConstructors? Constructs sounds to generic to me and I do not intuitively thought that the call of a constructor was meant.

Reply:
or `HasFunctionStyleCast` and `HasFunctionStyleConstruction`